src/lib/RoomConnection.ts

```diff
@@ -365,23 +374,17 @@ export default class RoomConnection extends TypedEventTarget {
         this.logger.log("Joining room");
         this.roomConnectionState = "connecting";
 
-        if (!this.localStream && this.localMediaConstraints) {
-            const localStream = await navigator.mediaDevices.getUserMedia(this.localMediaConstraints);
-            this.localStream = localStream;
-        }
-
-        const organization = await this.organizationServiceCache.fetchOrganization();
-        if (!organization) {
-            throw new Error("Invalid room url");
+        if (!this.localMedia.stream.active) {
+            await this.localMedia.start();
```

Comment:
At the moment, this could throw

Reply:
Will fix this separately, as this PR focuses more on the local media files. Before using localMedia in this class, the gUM could still throw, so this flaw has been here all along